zero/src/rpc/jerigon.rs

```diff
@@ -53,9 +116,75 @@ where
                 compact: hex::decode(block_witness.strip_prefix("0x").unwrap_or(&block_witness))
                     .context("invalid hex returned from call to eth_getWitness")?,
             }),
-            txn_info: tx_results.into_iter().map(|it| it.result).collect(),
+            txn_info,
             code_db: Default::default(),
         },
         other_data,
     })
 }
+
+/// Processes the transactions in the given block and updates the code db.
+pub async fn process_transactions<ProviderT, TransportT>(
```

Comment:
I don't think having two very similar public `process_transactions` (one in native one in jerigon) doing same work and returning different result  is a good idea. Could we unify the logic for both ?

Reply:
If we end up moving the fetching logic to Jerigon, this would become moot.

The idea was to reuse the same overall structure from Native but fetch exactly the data needed for Jerigon instead.  The alternative, if I understand you correctly, would be that each function includes a dispatching mechanism based on a tag and then return a tagged value.  I don't have a preference to either, but I would like some more inputs on this before rewriting it, @Nashtare @muursh.